testing/trino-testing/src/main/java/io/trino/testing/AbstractTestQueryFramework.java

```diff
@@ -555,6 +559,25 @@ protected OperatorStats searchScanFilterAndProjectOperatorStats(QueryId queryId,
                 .collect(MoreCollectors.onlyElement());
     }
 
+    private TableMetadata getTableMetadata(TableHandle tableHandle)
+    {
+        Session session = Session.builder(getSession())
+                .setCatalog(tableHandle.getCatalogName().getCatalogName())
+                .build();
+        return inTransaction(session, transactionSession -> getQueryRunner().getMetadata().getTableMetadata(transactionSession, tableHandle));
+    }
+
+    private <T> T inTransaction(Session session, Function<Session, T> transactionSessionConsumer)
+    {
+        return transaction(getQueryRunner().getTransactionManager(), getQueryRunner().getAccessControl())
+                .singleStatement()
+                .execute(session, transactionSession -> {
+                    // metadata.getCatalogHandle() registers the catalog for the transaction
+                    transactionSession.getCatalog().ifPresent(catalog -> getQueryRunner().getMetadata().getCatalogHandle(transactionSession, catalog));
+                    return transactionSessionConsumer.apply(transactionSession);
```

Comment:
You use `SessionBuilder.setCatalog` above to pass here information which catalog will be accessed

- pass it explicitly
- this may fail if query touches 2+ catalogs (eg "tested catalog join tpch"); the previous code didn't have this, right?